src/CMakeLists.txt

```diff
@@ -199,6 +199,14 @@ add_library(parthenon
   parthenon_mpi.hpp
 )
 
+if (BUILD_SHARED_LIBS)
+  add_library(parthenon SHARED "${parthenon_sources}")
+else()
+  add_library(parthenon STATIC "${parthenon_sources}")
+endif()
+
+set_target_properties(parthenon PROPERTIES SOVERSION 1)
```

Comment:
If I remember correctly, setting the SOVERSION will generate an error for BUILD_SHARED_LIBS=OFF.

Reply:
This doesn't seem to cause an error for me on cmake 3.17.1, but I'll go ahead and guard this call with a check for `BUILD_SHARED_LIBS=ON`